parcel-bundler/parcel
# Fix issue #2267

Fixes #2267 Compile to js doesn't modify script type=application/ts to
type=application/javascript



# ↪️ Pull Request

When a HTMLAsset script tag has a type different from text/javascript
and application/javascript, the HTMLAsset class does not rename the
type property of the tag to a valid application/javascript, and so this
fix forces that when a script tag whose sub asset has a detected a script of type js,
and has been transformed properly it will rewrite the type property of
the script tag to application/javascript.

Same issue happens with CSS, when a style tag with type property != to text/css is found
and it's sub asset is properly transformed then the final tag will include valid css and it's type
property will be transformed to a proper text/css type. 

## 💻 Examples



## 🚨 Test instructions

In order to test this fix the following steps must be done:
1. Create a simple html page with a style tag and/or a script tag.

1a. Style tag must have valid LESS or SCSS code and must include property type with the corresponding value (e.g. text/less, text/scss)

1b. Script tag must have valid typescript or elm or any other valid parcel supported code and must include property type with the corresponding value (e.g. application/typescript, application/elm...)

2. Run parcel

3. Check build folder configured and the script and/or the style tag type property should now be application/javascript and/or text/css
## ✔️ PR Todo
